src/dcegm/egm.py

```diff
@@ -151,10 +148,11 @@ def get_child_state_marginal_util_and_exp_max_value(
     child_node_choice_set = get_state_specific_choice_set(
```

Comment:
The child specific choice set, the choice_values_child and choice_policies_child are independent of the savings and income shocks. Can we determine them outside of the for loops and then use them as function argument?